phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -1887,6 +1899,13 @@ public boolean isImmutableRows() {
         return immutableRows;
     }
     
+    public boolean isViewIndex() {
+        if (parentTableType == null) {
```

Comment:
@sguggilam - In the upgrade scenario where the clients don't have the new proto and hence parentTableType isn't defined, will this still work OK to have isViewIndex incorrectly return true?

Reply:
Yes @gjacoby126 , the logic that is added in the change makes sure that the behavior is unchanged for the clients that doesn't pass the parentTableType field (the logic uses a OR condition with !isViewIndex() )